collector.go

```diff
@@ -39,18 +39,20 @@ func (c *collector) collect() (flatMetrics, error) {
 
 	m, err := cl.metrics()
 	if err != nil {
-		return nil, err
+		log.Printf("failed to get metrics: %v", err)
```

Comment:
I think we still should return after the log because `m` is probably `nil`.

Reply:
And it's totally fine that it's `nil`, because `transformMetrics` returns a `make(flatMetrics)`, ragardless of what `m` is:
https://github.com/status-im/geth_exporter/blob/60e2ce77d795c3df57c9b0eefd10ad24081d9f87/metrics.go#L33-L35
and:
https://github.com/status-im/geth_exporter/blob/60e2ce77d795c3df57c9b0eefd10ad24081d9f87/metrics.go#L37-L38
A `range` over a `nil` slice is like a `range` over an empty slice, so nothing happens.

Reply:
It's counterintuitive when reading the code but yeah, it might be fine and not cause panic. I would add a comment that `nil` is ok value.